File: network.py
import httpx
import requests
from fileio import FileIO

class NetworkHandler(object):
    print("connecting to api server ...")

    def __init__(self,
                 _auth_token: str,
                 _header: dict,
                 _proxy: dict,
                 _con_method: str = "GET"
                 ):
        self._auth = _auth_token
        self._header = _header
        self._proxy = _proxy
        self._con_method = _con_method.lower()
        self.api_url = "https://tapi.bale.ai/bot{}".format(self._auth)

    def makeMethod(self, method: str = "getMe"):
        return f"{self.api_url}/{method}"

    def create(self, method: str, params = None):
        app = httpx.Client(headers=self._header,
                           proxies=self._proxy,
                           params=params)
        
        if self._con_method == "get":
            data = app.get(self.makeMethod(method))
            if data.status_code == 200:
                if data.text.startswith("{") and data.text.endswith("}"):
                    return data.json()
                else:
                    return data.text
            else:
                if data.text.startswith("{") and data.text.endswith("}"):
                    return data.json()
                else:
                    return data.text

        elif self._con_method == "post":
            data = app.post(self.makeMethod(method))
            if data.status_code == 200:
                if data.text.startswith("{") and data.text.endswith("}"):
                    return data.json()
                else:
                    return data.text
            else:
                if data.text.startswith("{") and data.text.endswith("}"):
                    return data.json()
                else:
                    return data.text
                
    def upload(self, typeFile: str, filePath: str, chatid, caption, message_id):
        fio = FileIO(filePath)

        if not fio.existsStat():
            raise FileNotFoundError("Cannot Found file to send")
        
        if not fio.isFile():
            raise ValueError("Path is not File")
        
        params = {
            'chat_id': chatid,
            "caption": caption,
            "reply_to_message_id": message_id
        }

        file = {
            typeFile: fio.readStream()
        }

        try:
            return requests.get(self.makeMethod("sendPhoto"), params=params, files=file).json()
        except Exception as ER:
            return str(ER)

File: nodejs/network.js
const axiosClass = require("axios");
const FormData = require('form-data');
const fs = require('fs');

console.log("connecting to api server ...");

class NetworkHandler{
    constructor(
        auth_token,
        proxy,
        header){
        this.auth = auth_token;
        this.proxy = proxy;
        this.head = header;
        this.axGet = new axiosClass.Axios("GET");
        this.axPost = new axiosClass.Axios("POST");
    }

    makeMethod(method){
        return `https://tapi.bale.ai/bot${this.auth}/${method}`;
    }

    createGet(method, params, callback = null){
        try{
            this.axGet.get(this.makeMethod(method), {
                proxy: this.proxy,
                headers: this.head,
                params:params
            }).then((resp) => {
                if (callback === null){
                    true;
                }else{
                    let data = JSON.parse(resp.data);
                    data.error = false;
                    data.base = null;
                    callback(data)
                }
            })
        }catch (er){
            if (callback === null){
                true;
            }else{
                let data = JSON.parse(JSON.stringify(
                    {
                        "error": true,
                        "base": er
                    }
                ));
                callback(data)
            }
        }
    }

    createPost(method, params, callback = null){
        try{
            this.axPost.post(this.makeMethod(method), {
                proxy: this.proxy,
                headers: this.head,
                params:params
            }).then((resp) => {
                if (callback === null){
                    true;
                }else{
                    let data = JSON.parse(resp.data);
                    data.error = false;
                    data.base = null;
                    callback(data)
                }
            })
        }catch (er){
            if (callback === null){
                true;
            }else{
                let data = JSON.parse(JSON.stringify(
                    {
                        "error": true,
                        "base": er
                    }
                ));
                callback(data)
            }
        }
    }

    upload({TypeFile, path, chatID, caption = null, messageID = null, callback = null}={}){
        const formData = new FormData();
		const file = fs.createReadStream(path);
		formData.append(TypeFile, file);
		formData.append("chat_id", chatID);
		formData.append("caption", caption ? caption !== null : "");
		formData.append("reply_to_message_id", messageID ? messageID !== null : "")
		this.axPost.post(this.makeMethod(`send${TypeFile}`), formData, {
		headers: {
			'Content-Type': 'multipart/form-data',
		},
		})
		.then((response) => {
            if (callback === null){
                true;
            }else{
                let data = JSON.parse(response.data);
                data.error = false;
                data.base = null;
                callback(data);
            }
		})
		.catch((error) => {
			if (callback === null){
                true;
            }else{
                let data = JSON.parse(JSON.stringify({}));
                data.error = true;
                data.base = error;
                callback(data);
            }
		});
    }
}

module.exports = {
    NetworkHandler
}